src/Servers/Kestrel/Core/src/Internal/Http/Http1MessageBody.cs

```diff
@@ -159,6 +160,25 @@ protected async Task OnConsumeAsyncAwaited()
                 KestrelBadHttpRequestException.Throw(RequestRejectionReason.FinalTransferCodingNotChunked, transferEncoding);
             }
 
+            // https://datatracker.ietf.org/doc/html/rfc7230#section-3.3.2
+            // A sender MUST NOT send a Content-Length header field in any message
+            // that contains a Transfer-Encoding header field.
+            // https://datatracker.ietf.org/doc/html/rfc7230#section-3.3.3
+            // If a message is received with both a Transfer-Encoding and a
+            // Content-Length header field, the Transfer-Encoding overrides the
+            // Content-Length.  Such a message might indicate an attempt to
+            // perform request smuggling (Section 9.5) or response splitting
+            // (Section 9.4) and ought to be handled as an error.  A sender MUST
+            // remove the received Content-Length field prior to forwarding such
+            // a message downstream.
+            // We should remove the Content-Length request header in this case, for compatibility
+            // reasons, increase x-Content-Length so that the original Content-Length is still available.
+            if (headers.ContentLength.HasValue)
```

Comment:
I'm not quite sure if removing the request header addition is appropriate here, as it looks like this method is just for creating the MessageBody.